frontend/integration-tests/tests/kubevirt/clone.vm.scenario.ts

```diff
@@ -92,118 +88,115 @@ describe('Test clone VM.', () => {
+      await withResource(leakedResources, clonedVm.asResource(), async() => {
+        await clonedVm.navigateToListView();
 
-      await clonedVm.action('Delete');
-      leakedResources.delete(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
+        await filterForName(`${testContainerVm.metadata.name}-clone`);
+        await resourceRowsPresent();
+        await browser.wait(until.textToBePresentInElement(wizardView.firstRowVMStatus, 'Running'), VM_BOOTUP_TIMEOUT);
+        deleteResource(clonedVm.asResource());
+      });
     }, VM_BOOTUP_TIMEOUT);
 
     it('Cloned VM has cleared MAC addresses.', async() => {
       await vm.addNic(networkInterface);
       await vm.action('Clone');
       await wizard.next();
 
-      const clonedVm = new VirtualMachine(clonedVmConf);
-      leakedResources.add(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
-      await clonedVm.navigateToTab(TABS.NICS);
-
-      await browser.wait(until.and(waitForCount(resourceRows, 2)), PAGE_LOAD_TIMEOUT);
-      // TODO: Add classes/ids to collumn attributes so that divs can be easily selected
-      const mac = await resourceRows.first().$('div:nth-child(5)').getText();
-      expect(mac === emptyStr).toBe(true, 'MAC address should be cleared');
-
-      await clonedVm.action('Delete');
-      leakedResources.delete(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
+      await withResource(leakedResources, clonedVm.asResource(), async() => {
+        await clonedVm.navigateToTab(TABS.NICS);
+        await browser.wait(until.and(waitForCount(resourceRows, 2)), PAGE_LOAD_TIMEOUT);
+        const addedNic = (await clonedVm.getAttachedNics()).find(nic => nic.name === networkInterface.name);
+        expect(addedNic.mac === DASHES).toBe(true, 'MAC address should be cleared');
+        deleteResource(clonedVm.asResource());
+      });
       await vm.removeNic(networkInterface.name);
     }, VM_BOOTUP_TIMEOUT);
 
     it('Cloned VM has vm.kubevirt.io/name label.', async() => {
       await vm.action('Clone');
       await wizard.next();
 
-      const clonedVm = new VirtualMachine(clonedVmConf);
-      leakedResources.add(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
-      expect(searchYAML(`vm.kubevirt.io/name: ${vm.name}`, clonedVm.name, clonedVm.namespace, 'vm'))
-        .toBeTruthy('Cloned VM should have vm.kubevirt.io/name label.');
-
-      await clonedVm.action('Delete');
-      leakedResources.delete(JSON.stringify({name: clonedVm.name, namespace: clonedVm.namespace, kind: 'vm'}));
+      await withResource(leakedResources, clonedVm.asResource(), async() => {
+        expect(searchYAML(`vm.kubevirt.io/name: ${vm.name}`, clonedVm.name, clonedVm.namespace, 'vm'))
+          .toBeTruthy('Cloned VM should have vm.kubevirt.io/name label.');
+        deleteResource(clonedVm.asResource());
```

Comment:
is the reason for replacing `action('Delete')` with `deleteResource` only for separation of concern so that only one of our tests tests the `action('Delete')` ? Or is there any other reason?

Reply:
Yes, I think it's not necessary to delete all VMs via UI action since this has a test case for itself. So it may be better to save some time and delete it via CLI.
But I'm open to change it back if you think it's more valuable to keep the whole user workflow.

Reply:
I am okay with saving the time :) so long as it is tested at least in one place